sdk/program/src/lib.rs

```diff
@@ -1,3 +1,650 @@
+//!             vault_pda.clone(),
+//!         ],
+//!         &[
+//!             &[
+//!                 b"vault",
+//!                 payer.key.as_ref(),
+//!                 &[vault_bump_seed],
+//!             ],
+//!         ]
+//!     )?;
+//!     Ok(())
+//! }
+//! ```
+//!
+//! # Native programs
+//!
+//! Some solana programs are [_native programs_][np2], running native machine
+//! code that is distributed with the runtime, with well-known program IDs.
+//!
+//! [np2]: https://docs.solana.com/developing/runtime-facilities/programs
+//!
+//! Native programs are divided into several categories:
+//!
+//! __Built-ins__ act just like other Solana programs, in that they can
+//! be [invoked] from within Solana programs or from within [transactions]
+//! submitted by clients. They implement fundamental operations that require
+//! direct access to the runtime.
+//!
+//! [invoked]: #cross-program-instruction-execution
+//! [transactions]: https://docs.rs/solana-sdk/latest/solana_sdk/transaction/index.html
+//!
+//! __Special built-ins__ include a few programs that are defined like built-ins,
+//! but do not work when invoked from another program, instead only functioning
+//! when submitted as part of a transaction. The exact behavior when invoked
+//! from another program depends on the built-in.
+//!
+//! __Precompiles__ are extremely compute-intensive operations. These can not be
+//! invoked by other programs, and can only be executed as instructions within
+//! transactions submitted by clients. Precompile instructions may be
+//! batch-executed in parallel with other precompiles prior to execution of
+//! other instructions. Invoking a precompile from another program will return
+//! an error.
+//!
+//! __Program loaders__ are responsible for loading and executing Solana programs.
+//! There may be multiple program loaders. Today the only program loaders are
+//! for [eBPF] programs. When an executable program's ownership is assigned to a
+//! particular program loader then that loader will be used to run the program.
+//! Program loaders are typically invoked by the CLI `solana deploy` command,
+//! to deploy programs, and by the runtime to execute programs. Invoking a
+//! program loader from another program will return an error.
+//!
+//! This crate defines the program IDs for most native programs: even though
+//! some native programs cannot be invoked by other programs, a Solana program
+//! may need to verify that certain accounts represent those native programs,
+//! e.g. when verifying that the ed25519 precompile verified a signature.
+//!
+//! For many built-ins, this crate also defines enums that represent the
+//! instructions they process, and constructors for building the instructions.
+//! Some built-ins' instructions and constructors, though technically callable
+//! from Solana programs, are not defined somewhere accessible to Solana
+//! programs. This is a historical artifact due to lack of use cases for
```

Comment:
Which are you referring to, maybe we should move then into this crate

Reply:
From my reading, these native programs could be invoked from other programs but their instructions are defined elsewhere:

- solana_vote_program
- solana_config_program
- address_lookup_table_program

Reply:
Programs don't need to be visible to other programs as programs won't/shouldn't be calling them.